File: rtllm/StructClass/InferenceRequest.py
class InferenceRequest:
    def __init__(self,createCacheMode:bool,input,historyFileName:str,cacheFilename:str=None,clientSocket=None,addr=None,requestID:int=None,priority:int=0,prevOutput=None):
        self.createCacheMode = createCacheMode
        self.input = input
        self.historyFileName = historyFileName
        self.cacheFilename = cacheFilename
        self.clientSocket = clientSocket
        self.addr = addr
        self.requestID = requestID
        self.priority = priority
        self.prevOutput = prevOutput
        # print('InferenceRequest ',input)

    def __copy__(self):
        return InferenceRequest(createCacheMode=self.createCacheMode,
                                input=self.input,
                                historyFileName=self.historyFileName,
                                cacheFilename=self.cacheFilename,
                                clientSocket=self.clientSocket,
                                addr=self.addr,
                                requestID=self.requestID,
                                priority=self.priority,
                                prevOutput=self.prevOutput)

File: rtllm_cpp/Structures/InferenceRequest.cpp
#include <iostream>
#include <string>

using namespace std;
class InferenceRequest
{
    private:
        bool createCacheMode;
        string input;
        string cacheFileName;
        int client;
        string addr;
        int requestID;
        int priority;
        int prevOutput;
    public :
        InferenceRequest(bool createCacheMode,string input,string cacheFileName, int client, string addr, int requestID, int priority, int prevOutput) {
            createCacheMode= createCacheMode
        }
}